hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/TenantControl.java

```diff
@@ -65,5 +64,29 @@
      * Hazelcast object from the tenant
      * This is so the TenantControl itself can be garbage collected
      */
-    void unregister();
+    void distributedObjectDestroyed();
```

Comment:
Minor: maybe we can change the method names `distributedObjectCreated` and `distributedObjectDestroyed`. I understand they should be called when the object is created and destroyed but this tells more about _when_ they will be called rather than _what_ the implementation should do. In some other case, maybe we might call these methods when the object hasn't yet been created or destroyed.

Maybe something like `registerContext` or `unregisterContext` is better?

Reply:
renamed to registerObject / unregisterObject